dmsa/models.py

```diff
@@ -0,0 +1,65 @@
+import cPickle as pickle
+import logging
+import simpleflock
+
+LOCK_FILE = 'dmsa.models.lockfile'
+MODELS_FILE = 'dmsa.models'
+RESOURCE_TEMPORARILY_UNAVAILABLE = 35  # An IOError errno
+
+
+def _pickle_and_cache_models(models):
+    try:
+        with open(MODELS_FILE, mode='w') as f:
+            try:
+                pickle.dump(models, f)
+            except pickle.PicklingError as e:
+                logging.error('pickling data models: {}'.format(e))
+                raise
+    except IOError as e:
+        logging.error('opening {} for writing: {}'.format(MODELS_FILE, e))
+        raise
```

Comment:
Exceptions can be stacked. Most context managers (for `with`) don't swallow exceptions.

```python
try:
    ...
except pickle.PicklingError as e:
    ...
except IOError as e:
    ...
```

But a more general question, what is the value of logging these errors? Where do these log statements go, can they be viewed and/or alerted, and are they actionable?

Reply:
We should probably adopt the convention that all of our exceptions and logging should go to "the" dbhi sentry server unless there is something more specific arranged. Is there a communal Sentry server?  I was thinking we could add Sentry support in a subsequent PR.

Reply:
And thanks for the reminder about batching the exception handling.